### PR TITLE
feat: add dark mode toggle button

### LINKED ISSUE #22: feat: add dark mode switcher
https://tailwindcss.com/docs/dark-mode

### PR DESCRIPTION
**Pull request by ssarkar551**
feat: add dark mode toggle button

### CODE REVIEW

**Comment by AyanavaKarmakar** [comment]
@ssarkar551 resolve merge conflicts. Then update the branch, since changes from #14 are leaking in

**Comment by ssarkar551** [comment]
resolved merge conflicts

File: package.json [hunk 1]
```diff
@@ -35,8 +35,9 @@
     "@trpc/next": "^10.7.0",
     "@trpc/react-query": "^10.7.0",
     "@trpc/server": "^10.7.0",
-    "next": "13.1.1",
+    "next": "^13.1.1",
```
Comment (AyanavaKarmakar) [inline package.json]: Check if the behaviour remains the same after removing the `^`

```suggestion
    "next": "13.1.1",
```

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -4,15 +4,18 @@ import { SessionProvider } from "next-auth/react";
...
 
 const MyApp: AppType<{ session: Session | null }> = ({
   Component,
   pageProps: { session, ...pageProps },
 }) => {
   return (
-    <SessionProvider session={session}>
-      <Component {...pageProps} />
-    </SessionProvider>
+    <ThemeProvider defaultTheme="light">
```
Comment (AyanavaKarmakar) [inline src/pages/_app.tsx]: Set the default theme to dark. Let's not burn the user's eye with the bright light theme if they open the site at night.

```suggestion
    <ThemeProvider defaultTheme="dark">
```

File: src/styles/globals.css [hunk 3]
```diff
@@ -7,6 +7,10 @@
     /** reverts back to custom font after loading system font on slower connections  */
     font-display: swap;
     /** TODO better background color */
-    @apply bg-black;
+
+    /* @apply bg-black; */
```
Comment (AyanavaKarmakar) [inline src/styles/globals.css]:
```suggestion
```

Comment (AyanavaKarmakar) [reply src/styles/globals.css]: Remove the extra space too.

image

File: src/styles/globals.css [hunk 4]
```diff
@@ -7,6 +7,10 @@
     /** reverts back to custom font after loading system font on slower connections  */
     font-display: swap;
     /** TODO better background color */
-    @apply bg-black;
+
```
Comment (AyanavaKarmakar) [inline src/styles/globals.css]:
```suggestion
```

File: src/pages/index.tsx [hunk 5]
```diff
@@ -31,6 +36,38 @@ const Home: NextPage = () => {
             </p>
           </div>
         </Link>
+
+        <button
```
Comment (AyanavaKarmakar) [inline src/pages/index.tsx]: create a components folder in src. inside that create a file called ThemeToggle.tsx and move this button there (/src/components/ThemeToggle.tsx)

Now, create an index.ts file inside the components folder and export the component there like this ->

```ts
export { ThemeToggle } from "./ThemeToggle";
```

Then import it here like this ->
```tsx
import { ThemeToggle } from "./components"
```

This is for component reusability

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]
Run  `npm run format`. Then, do the changes. Before pushing the code, run it again. The extra space issues will be fixed by running the script probably.

File: src/pages/index.tsx [hunk 6]
```diff
@@ -31,6 +36,38 @@ const Home: NextPage = () => {
...
+          id="theme-toggle"
+          type="button"
+          className="rounded-lg p-2.5 text-sm text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-4 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-700"
+          onClick={() => {setTheme(theme === "dark"? "light" : "dark")}}
+        >
+          <svg
```
Comment (AyanavaKarmakar) [inline src/pages/index.tsx]: Where is the svg being used here? 

image

**Comment by AyanavaKarmakar** [comment]
Re-request review after applying the changes @ssarkar551

File: src/pages/index.tsx (hunk 6 above)
Comment (ssarkar551) [reply src/pages/index.tsx]: Should be showing a light theme and dark theme icon, but not being able to. Trying to figure it out.

Comment (AyanavaKarmakar) [reply src/pages/index.tsx]: @ssarkar551 I think we should remove the SVG entirely, cuz we will be swapping the design with radix UI later

Comment (ssarkar551) [reply src/pages/index.tsx]: Okay

File: src/pages/index.tsx (hunk 5 above)
Comment (ssarkar551) [reply src/pages/index.tsx]: applied @AyanavaKarmakar

**Review by Nishith-Savla — APPROVED** [review]

**Review by AyanavaKarmakar — APPROVED** [review]
LGTM

**Pull request merged**